data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/MutateMapper.java

```diff
@@ -31,21 +31,36 @@ public AddEntryConfig(String key, Object value) {
         public final String from_key;
         public final String to_key;
 
-        public RenameCopyConfig(String fromKey, String toKey) {
+        public RenameCopyConfig(final String fromKey, final String toKey) {
             this.from_key = fromKey;
             this.to_key = toKey;
         }
     }
 
-    public List<PluginModel> mapAttributes(List<LogstashAttribute> logstashAttributes, LogstashAttributesMappings logstashAttributesMappings) {
-        List<PluginModel> models = new LinkedList<>();
-        List<AddEntryConfig> adds = new LinkedList<>();
-        List<RenameCopyConfig> renames = new LinkedList<>();
-        List<String> deletes = new LinkedList<>();
-        List<RenameCopyConfig> copies = new LinkedList<>();
-        List<String> uppercases = new LinkedList<>();
+    public static class SubstituteConfig {
+        public final String source;
+        public final String from;
+        public final String to;
 
-        for(LogstashAttribute attr : logstashAttributes) {
+        public SubstituteConfig(final String source, final String from, final String to) {
+            this.source = source;
+            this.from = from;
+            this.to = to;
+        }
+    }
+
+    public List<PluginModel> mapAttributes(final List<LogstashAttribute> logstashAttributes, final LogstashAttributesMappings logstashAttributesMappings) {
+        final List<PluginModel> models = new LinkedList<>();
+        final List<AddEntryConfig> adds = new LinkedList<>();
+        final List<RenameCopyConfig> renames = new LinkedList<>();
+        final List<String> deletes = new LinkedList<>();
+        final List<RenameCopyConfig> copies = new LinkedList<>();
+        final List<String> uppercases = new LinkedList<>();
+        final List<String> lowercases = new LinkedList<>();
+        final List<String> trims = new LinkedList<>();
+        final List<SubstituteConfig> substitutes = new LinkedList<>();
```

Comment:
That's a lot of LinkedLists! I am looking forward to when we redesign this class after we get these features out.